src/core/evaluator.js

```diff
@@ -1030,8 +1033,22 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
       function runBidiTransform(textChunk) {
         var str = textChunk.str.join('');
         var bidiResult = PDFJS.bidi(str, -1, textChunk.vertical);
+        var shouldReplaceWhitespace = false;
+
+        if (normalizeWhitespace) {
+          // Replace all whitespaces with standard spaces (0x20), to avoid
+          // alignment issues between the textLayer and the canvas if the text
+          // contains e.g. tabs (fixes issue6612.pdf).
+          var i = 0, ii = bidiResult.str.length, charCode;
+          while (i < ii && (charCode = bidiResult.str.charCodeAt(i)) >= 0x20 &&
+                 charCode <= 0x7F) {
+            i++;
+          }
+          shouldReplaceWhitespace = i < ii;
```

Comment:
Create a function that will handle the whitespace replacement? Please notice to be compiled function must be short and invoke lots of times. The inlining may not produce compiled result hence make loop with charCodeAt show.